File: src/main/java/com/github/unchama/menuinventory/InventorySize.kt
package com.github.unchama.menuinventory

import arrow.core.Either
import arrow.core.left
import org.bukkit.event.inventory.InventoryType

/**
 * チェストインベントリの行数を表すdata class
 */
data class InventoryRowSize(val rows: Int)

/**
 * インベントリのサイズを表すデータ型.
 */
typealias InventorySize = Either<InventoryRowSize, InventoryType>

fun Int.rows(): InventorySize = InventoryRowSize(this).left()

File: src/main/java/com/github/unchama/menuinventory/InventorySize.scala
package com.github.unchama.menuinventory

import org.bukkit.event.inventory.InventoryType
/**
 * チェストインベントリの行数を表すcase class
 */
case class InventoryRowSize(val rows: Int)
object InventoryRowSize {
  /**
   * インベントリのサイズを表すデータ型.
   */
  type InventorySize = Either[InventoryRowSize, InventoryType]

  implicit class IntToInventorySize(val int: Int) {
    def rows: InventorySize = Left(InventoryRowSize(int))
  }
}
